MyStore/MyStore.cpp

```diff
@@ -1,12 +1,38 @@
-﻿#include "JsonParcer.h"
+﻿#include "StorageData.h"
+#include <conio.h>
+#include <chrono>
 
+const int period_for_element_milliseconds = 1;
+using namespace std::chrono;
 
-int main(int argc, char *argv[])
+std::ostream& operator << (std::ostream& out, const InfoProduct& data)
 {
+	out << "Name:" << data.m_product_color << '\n' 
+		<< "Color:" << data.m_product_name << '\n';
+	return out;
+}
 
-	JsonParcer obj("..\\MyStore\\data\\Data.json");
-	obj.Deserialization();
-	obj.ShowData();
+
+int main(int argc, char* argv[])
+{
+	StorageData storage;
+
+	storage.m_Parcer.Deserialization();
+	storage.m_Parcer.ShowData();
+
+	auto last_time = system_clock::now();
+
+	while (true)
+	{
+		if ((duration_cast<seconds>(system_clock::now() - last_time).count()) > period_for_element_milliseconds)
+		{
+			std::cout << storage.m_Generator.GeneratedProduct();
+			last_time = system_clock::now();
+		}
+		char c = _getche();
+		if (c == ' ')

```

Comment:
Great, now code almost makes what it should! But here's the problem. _getche() waits for pressing any key and until you do that, UI thread is blocked. Therefore if you run program and don't press any keys, products will not generate... To avoid this I'd suggest to use the following condition:

		if (GetKeyState(VK_SPACE) & 0x8000)
			break;

It just checks the state of specific key without blocking UI thread which doesn't block program execution. To use this code you need to include <windows.h>